backend/tests/models/user.test.ts

```diff
@@ -29,168 +28,404 @@ describe('User model and validation tests', () => {
+        callback(null, null);
+      });
+
+      const result = await User.checkEmail('nonexistentemail@fakesite.com');
+      expect(result).toBe(false);
+      expect(findByUsernameMock).toHaveBeenCalledWith('nonexistentemail@fakesite.com', false, expect.any(Function));
+    });
+
+    it('should reject if an error occurs during the search', async () => {
+      findByUsernameMock = jest.spyOn(User, 'findByUsername').mockImplementation((email, select, callback) => {
+        callback(new Error('Database error'), null);
+      });
+
+      await expect(User.checkEmail('error@fakesite.com')).rejects.toThrow('Database error');
+      expect(findByUsernameMock).toHaveBeenCalledWith('error@fakesite.com', false, expect.any(Function));
+    });
   });
 
-  it.skip('sends an email with nodemailer', async () => {
-    // Skipping test because kinda pointless and brittle,
-    // But it's a nice example if you want to test other email methods
-    const sendMailFn = jest.fn();
-    mockedNodemailer.createTransport.mockReturnValue(({ sendMail: sendMailFn } as unknown) as Transporter);
-    const content = {
-      to: '',
-      subject: '',
-      text: ''
-    };
-    const mockUser = new User();
-
-    await mockUser.sendMail(content);
+  describe('Token generation methods', () => {
+    it('generates a password reset token, sets the resetPasswordToken field, and sets expiry time', async () => {
+      const user = new User({ email: 'test@example.com' });
+      await user.generatePasswordResetToken();
+      expect(user.resetPasswordToken).toBeDefined();
+      expect(user.resetPasswordExpires).toBeInstanceOf(Date);
+    });
+
+    it('generates an email verification token, sets the verifyEmailToken field, and sets expiry time', async () => {
+      const user = new User({ email: 'test@example.com' });
+      await user.generateEmailVerificationToken();
+      expect(user.verifyEmailToken).toBeDefined();
+      expect(user.verifyEmailTokenExpires).toBeInstanceOf(Date);
+    });
+
+    it('generates a beta access token, sets the betaAccessToken field, and sets expiry time', async () => {
+      const user = new User({ email: 'test@example.com' });
+      await user.generateBetaAccessToken();
+      expect(user.betaAccessToken).toBeDefined();
+      expect(user.betaAccessTokenExpires).toBeInstanceOf(Date);
+    });
+  });
 
-    expect(nodemailer.createTransport).toHaveBeenCalled();
-    expect(sendMailFn).toHaveBeenCalled();
+  describe('Email sending methods', () => {
+    let sendMailMock: jest.Mock;
+
+    beforeEach(() => {
+      jest.clearAllMocks();
+      sendMailMock = jest.fn().mockResolvedValue({});
+    });
+  
+    it('sends an email with nodemailer', async () => {
+      const sendMailFn = jest.fn();
+      mockedNodemailer.createTransport.mockReturnValue(({ sendMail: sendMailFn } as unknown) as Transporter);
+      const content = {
+        from: '"The CDJ Team" <system@thecdj.app>',
```

Comment:
I'm not seeing how it's reading from an env variable. Would you explain how it is if sendMail is mocked?

Reply:
Ah there's 2 methods named `sendMail()`; that's what's confusing. You mocked out the `nodemailer.Transporter.sendMail()` method, so line 310 in models/user.ts will be the mock function. But the `User.sendMail()` method isn't mocked out because it's what you're testing, and it reads from the environment variables to create the message that gets sent. The code in models/user.ts is as follows: `  const message = {
    from: `"${process.env.SMTP_NAME}" <${process.env.SYSTEM_INBOX}>`,
    to,
    subject,
    text,
  };`

This is the message that you're testing for a match. Hard coding the from property is causing this test to break when a different developer changes these variable values. Hope that clarifies things!

Reply:
Oh yeah that is confusing. I think we should rename that method at some point. I'll make an issue for it. I think I resolved it though. I'll push it up once I address your other reviews.